lib/macho/macho_file.rb

```diff
@@ -397,11 +431,18 @@ def add_rpath(path)
     #  file.delete_rpath("/lib")
     #  file.rpaths # => []
     # @param path [String] the runtime path to delete
+    # @param options [Hash]
+    # @option options [Boolean] :strict (true) whether or not to fail loudly
+    #  with an exception if the change cannot be performed
     # @return void
     # @raise [MachO::RpathUnknownError] if no such runtime path exists
-    def delete_rpath(path)
+    def delete_rpath(path, options = {})
       rpath_cmds = command(:LC_RPATH).select { |r| r.path.to_s == path }
-      raise RpathUnknownError.new(path) if rpath_cmds.empty?
+
+      if rpath_cmds.empty?
+        return unless options.fetch(:strict, true)
+        raise RpathUnknownError.new(path)
+      end
```

Comment:
This change needs to be reverted. The modification methods in `MachOFile` need to always raise an exception, independently of `options` (because `options` currently only matters if multiple Mach-O slices are involved).